### ISSUE TITLE
`sell_fill` Telegram message gets send twice

### ISSUE THREAD

**Issue by Theagainmen** [freqtrade/freqtrade#5989]
## Describe your environment

  * Operating system: Debian x11
  * Python Version: 3.9.2 
  * CCXT version: 1.61.92
  * Freqtrade Version: 2021.11
  

## Describe the problem:

The sell fill message gets send twice, see screenshot. I only expect it to be send one time, once the order gets fulfilled.
![image](https://user-images.githubusercontent.com/24569139/143784929-10d1d1b2-64ca-44a7-b8d8-0ed1f55cda69.png)

### Steps to reproduce:

  1. Set `sell_fill` to `on` in the Telegram config part
  2. Wait for your bot to make a trade and fill it, I am running live (maybe that matters)
  
### Observed Results:

  * What happened?
The fill sell message got sent twice.

  * What did you expect to happen?
I expect the fill sell message to only send once.

### Relevant code exceptions or logs

As you can see in the logs, it also says here that it sends the sell_fill message twice.

  ```
2021-11-28 17:52:26,512 - freqtrade.wallets - INFO - Wallets synced.
2021-11-28 17:52:26,550 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=2021-11-28 16:26:47)
2021-11-28 17:52:26,580 - freqtrade.freqtradebot - INFO - Applying fee on amount for Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=2021-11-28 16:26:47) failed: myTrade-Dict empty found
2021-11-28 17:52:26,580 - freqtrade.persistence.models - INFO - Updating trade (id=1) ...
2021-11-28 17:52:26,580 - freqtrade.persistence.models - INFO - LIMIT_SELL has been fulfilled for Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=2021-11-28 16:26:47).
2021-11-28 17:52:26,581 - freqtrade.persistence.models - INFO - Marking Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.
2021-11-28 17:52:26,585 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell_fill, 'trade_id': 1, 'exchange': 'Gateio', 'pair': 'REQ/USDT', 'gain': 'profit', 'limit': 0.56028, 'order_type': '', 'amount': 35.883, 'open_rate': 0.54161, 'close_rate': 0.56028, 'current_rate': None, 'profit_amount': 0.5987652, 'profit_ratio': 0.03075389, 'buy_tag': '3 ', 'sell_reason': 'sell_profit_w_1_51 ( 3 )', 'open_date': datetime.datetime(2021, 11, 28, 16, 26, 47, 560269), 'close_date': datetime.datetime(2021, 11, 28, 17, 52, 26, 581087), 'stake_currency': 'USDT', 'fiat_currency': 'EUR'}
2021-11-28 17:52:26,865 - freqtrade.wallets - INFO - Wallets synced.
2021-11-28 17:52:26,872 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell, 'trade_id': 1, 'exchange': 'Gateio', 'pair': 'REQ/USDT', 'gain': 'profit', 'limit': 0.56028, 'order_type': 'limit', 'amount': 35.883, 'open_rate': 0.54161, 'close_rate': 0.56028, 'current_rate': 0.54923, 'profit_amount': 0.5987652, 'profit_ratio': 0.03075389, 'buy_tag': '3 ', 'sell_reason': 'sell_profit_w_1_51 ( 3 )', 'open_date': datetime.datetime(2021, 11, 28, 16, 26, 47, 560269), 'close_date': datetime.datetime(2021, 11, 28, 17, 52, 26, 581087), 'stake_currency': 'USDT', 'fiat_currency': 'EUR'}
2021-11-28 17:52:27,132 - freqtrade.wallets - INFO - Wallets synced.
2021-11-28 17:52:27,144 - freqtrade.freqtradebot - INFO - Pair REQ/USDT is still locked until 2021-11-28 17:55:00 due to Auto lock.
2021-11-28 17:52:33,318 - freqtrade.freqtradebot - INFO - Updating sell-fee on trade Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=closed) for order 98283895396.
2021-11-28 17:52:33,319 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=closed)
2021-11-28 17:52:33,343 - freqtrade.persistence.models - INFO - Updating trade (id=1) ...
2021-11-28 17:52:33,344 - freqtrade.persistence.models - INFO - Marking Trade(id=1, pair=REQ/USDT, amount=35.88300000, open_rate=0.54161000, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.
2021-11-28 17:52:33,349 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': sell_fill, 'trade_id': 1, 'exchange': 'Gateio', 'pair': 'REQ/USDT', 'gain': 'profit', 'limit': 0.56028, 'order_type': '', 'amount': 35.883, 'open_rate': 0.54161, 'close_rate': 0.56028, 'current_rate': None, 'profit_amount': 0.5987652, 'profit_ratio': 0.03075389, 'buy_tag': '3 ', 'sell_reason': 'sell_profit_w_1_51 ( 3 )', 'open_date': datetime.datetime(2021, 11, 28, 16, 26, 47, 560269), 'close_date': datetime.datetime(2021, 11, 28, 17, 52, 26, 581087), 'stake_currency': 'USDT', 'fiat_currency': 'EUR'}
2021-11-28 17:52:33,629 - freqtrade.wallets - INFO - Wallets synced.
  ```

**Comment by waresz** [comment]
Same problem, although this does not affect the result

**Comment by xmatthias** [comment]
it's something I've not seen yet - and also never experienced (also not on gateio) - but then i don't really use fill notifications either.
That said - the code is also not suggesting why this should be happening except there's something fluky going on with the database ... 

you can disable these if you're bothered by that - but i am sure you know that  - and that's not the point of this issue :laughing: 

what's your database say about this particular trade? (both the trade and orders table, please)

**Comment by Theagainmen** [comment]
>it's something I've not seen yet
> you can disable these if you're bothered by that - but i am sure you know that - and that's not the point of this issue 😆

I get it every now and then, it doesn't bother me too much, but there's something going on there which might lead to another bug maybe :) ?

Not sure how I can copy paste it nicely, so here we go from the `orders database`:
```
1	1	buy	REQ/USDT	0	98262910655	closed	REQ/USDT	limit	buy	0.54161	0.54161	35.948	35.948	0.0	19.46979628	2021-11-28 16:26:47.566000	2021-11-28 16:26:56.155602	2021-11-28 16:26:56.155608
4	1	sell	REQ/USDT	0	98283895396	closed	REQ/USDT	limit	sell	0.54923	0.56028	35.883	35.883	0.0	20.10452724	2021-11-28 17:52:26.556000	2021-11-28 17:52:33.331761	2021-11-28 17:52:33.331764
```

From the `trades database`:
```
1	gateio	REQ/USDT	0	0.0018	0.0647064	REQ	0.0018	0.036188149032	USDT	0.54161	0.54161	19.469573894934	0.56028	0.54923	0.03075389	0.5987652	19.47	35.883	35.948	2021-11-28 16:26:47.560269	2021-11-28 17:52:26.581087		0.00541610000000001	-0.99	0.00541610000000001	-0.99		2021-11-28 16:26:56.236017	0.55463	0.5268	sell_profit_w_1_51 ( 3 )	closed	XXXX_live_gateio	3 	5
```

Sorry but I can't figure out how to copy paste the column names into here too..

**Comment by xmatthias** [comment]
kindof confirms the log - it's quite simple, the order is immediately filled and passes the same method twice, once as "answer from create_order" - and then a 2nd time to update fees (which were missing from the initial response).

Not really something surprising - just the notification is kinda odd/bothering - but should be rather simple to fix.
